### PR TITLE
Base Api

### PR DESCRIPTION
**Pull request by ntmdev116**
## Related Tickets
- [#Task 61250: Initial API](https://edu-redmine.sun-asterisk.vn/issues/61250)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/sun/imageeditor/data/model/PhotoCollection.kt [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
 
 data class PhotoCollection(
-    val id: String,
-    val title: String,
-    val coverPhotoUrl: String,
-    val photoUrlList: MutableList<String>)
+    var id: String = "",
+    var title: String = "",
+    var coverPhotoUrl: String = "",
+)
```
Comment (minhnn-1279) [inline app/src/main/java/com/sun/imageeditor/data/model/PhotoCollection.kt]: vẫn val được nhé :v mình sài copy()

File: app/src/main/java/com/sun/imageeditor/data/repository/source/remote/fetchjson/GetJsonFromUrl.kt [hunk 2]
```diff
@@ -1,4 +1,83 @@
...
+import com.sun.imageeditor.data.repository.source.OnResultListener
+import com.sun.imageeditor.utils.ApiConstant
+import org.brotli.dec.BrotliInputStream
+import java.io.FileNotFoundException
+import java.net.HttpURLConnection
+import java.net.URL
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+
+class GetJsonFromUrl<T> constructor(
```
Comment (minhnn-1279) [inline app/src/main/java/com/sun/imageeditor/data/repository/source/remote/fetchjson/GetJsonFromUrl.kt]:
```suggestion
class GetJsonFromUrl<T>(
```

File: app/src/main/java/com/sun/imageeditor/data/repository/source/remote/fetchjson/GetJsonFromUrl.kt [hunk 3]
```diff
@@ -1,4 +1,83 @@
...
+        val url = URL(urlString)
+        val httpURLConnection = url.openConnection() as? HttpURLConnection
+        httpURLConnection?.run {
+            connectTimeout = TIME_OUT
+            readTimeout = TIME_OUT
+            requestMethod = METHOD_GET
+            doInput = true
+            setRequestProperty("Authorization", "Client-ID ${ApiConstant.BASE_API_KEY}")
+            setRequestProperty("Accept", "*/*")
+            setRequestProperty("Accept-Encoding", "gzip, deflate, br")
```
Comment (minhnn-1279) [inline app/src/main/java/com/sun/imageeditor/data/repository/source/remote/fetchjson/GetJsonFromUrl.kt]: string resource hoặc constant cx đc nhé :v

**Review by minhnn-1279 — APPROVED** [review]

File: app/build.gradle [hunk 4]
```diff
@@ -44,6 +44,10 @@ dependencies {
     implementation 'com.google.android.material:material:1.8.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
     implementation 'com.github.bumptech.glide:glide:4.13.0'
+    implementation 'com.squareup.okhttp3:okhttp:3.14.7'
+    implementation 'org.brotli:dec:0.1.2'
+    implementation 'com.google.code.gson:gson:2.8.9'
```
Comment (ghost) [inline app/build.gradle]: dùng httpUrlConnection và tự parse json.

**Review by ghost — APPROVED** [review]

**Pull request merged**